src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/Checksum.java

```diff
@@ -31,6 +31,12 @@ private InvalidChecksumException(String msg) {
     }
   }
 
+  public static final class ICExceptionExtended extends Exception {
```

Comment:
nit: it's generally best practice to avoid abbreviations in class names unless they're standard. The reader may not know what "IC" stands for here. Also "extended" makes it sound like this exception extends another exception, but it doesn't. Maybe "MismatchedChecksumException" would be a better name?